clusterloader2/framework/util/util.go

```diff
@@ -0,0 +1,201 @@
+	if _, shouldRetry := apierrs.SuggestsClientDelay(err); shouldRetry {
+		return true
+	}
+	return false
+}
+
+// CreateNamespace creates a single namespace with given name.
+func CreateNamespace(c clientset.Interface, namespace string) error {
+	createFunc := func() (bool, error) {
+		_, err := c.CoreV1().Namespaces().Create(&apiv1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}})
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to create namespace with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(createFunc)
+
+}
+
+// DeleteNamespace deletes namespace with given name.
+func DeleteNamespace(c clientset.Interface, namespace string) error {
+	deleteFunc := func() (bool, error) {
+		err := c.CoreV1().Namespaces().Delete(namespace, nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to delete namespace with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(deleteFunc)
+}
+
+// CreateObject creates object based on given object description.
+func CreateObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind, obj *unstructured.Unstructured) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	obj.SetName(name)
+	createFunc := func() (bool, error) {
+		_, err := dynamicClient.Resource(gvr).Namespace(namespace).Create(obj)
+		if err == nil || apierrs.IsAlreadyExists(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to create object with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(createFunc)
+}
+
+// DeleteObject deletes object with given name, group, version and kind.
+func DeleteObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind) error {
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	deleteFunc := func() (bool, error) {
+		err := dynamicClient.Resource(gvr).Namespace(namespace).Delete(name, nil)
+		if err == nil || apierrs.IsNotFound(err) {
+			return true, nil
+		}
+		if IsRetryableAPIError(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("Failed to delete object with non-retryable error: %v", err)
+	}
+	return RetryWithExponentialBackOff(deleteFunc)
+}
+
+// GetObject retrieves object with given name, group, version and kind.
+func GetObject(dynamicClient dynamic.Interface, namespace string, name string, gvk schema.GroupVersionKind) (*unstructured.Unstructured, error) {
+	var obj *unstructured.Unstructured
+	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	getFunc := func() (bool, error) {
+		var err error
+		obj, err = dynamicClient.Resource(gvr).Namespace(namespace).Get(name, metav1.GetOptions{
+			IncludeUninitialized: true,
```

Comment:
It's not clear to me that we want to return uninitialized objects. In fact, I think we shouldn't.

So let's remove this and add a TODO to investigate if we have a need to include uninitialized objects.

Reply:
TODO added.